backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java

```diff
@@ -1695,6 +1699,218 @@ void testSearchWithType() {
         assertEquals(0, endpointPage.getData().size());
     }
 
+    /**
+     * Tests that when an invalid URL is provided via the endpoint's properties, regardless if those properties are
+     * {@link CamelProperties} or {@link WebhookProperties}, the proper constraint violation message is returned from
+     * the handler.
+     */
+    @Test
+    void testEndpointInvalidUrls() {
+        // Set up the RBAC access for the test.
+        final String orgId = "endpoint-invalid-urls";
+        final String userName = "user";
+
+        final String identityHeaderValue = TestHelpers.encodeRHIdentityInfo(orgId, orgId, userName);
+        final Header identityHeader = TestHelpers.createRHIdentityHeader(identityHeaderValue);
+
+        MockServerConfig.addMockRbacAccess(identityHeaderValue, MockServerConfig.RbacAccess.FULL_ACCESS);
+
+        // Set up the fixture data.
+        final var disableSslVerification = false;
+        final HttpType method = HttpType.POST;
+        final String password = "endpoint-invalid-urls-basic-authentication-password";
+        final String username = "endpoint-invalid-urls-basic-authentication-username";
+        final String secretToken = "endpoint-invalid-urls-secret-token";
+
+        // Create the properties for the endpoint. Leave the URL so that we can set it afterwards.
+        final var camelProperties = new CamelProperties();
+        camelProperties.setBasicAuthentication(new BasicAuthentication(username, password));
+        camelProperties.setDisableSslVerification(disableSslVerification);
+        camelProperties.setSecretToken(secretToken);
+
+        final var webhookProperties = new WebhookProperties();
+        webhookProperties.setBasicAuthentication(new BasicAuthentication(username, password));
+        webhookProperties.setDisableSslVerification(disableSslVerification);
+        webhookProperties.setMethod(method);
+        webhookProperties.setSecretToken(secretToken);
+
+        // Create an endpoint without the type and the properties set.
+        final var name = "endpoint-invalid-urls-name";
+        final var description = "endpoint-invalid-urls-description";
+        final var enabled = true;
+        final var serverErrors = 0;
+        final var subType = "slack";
+
+        final var endpoint = new Endpoint();
+        endpoint.setDescription(description);
+        endpoint.setEnabled(enabled);
+        endpoint.setName(name);
+        endpoint.setServerErrors(serverErrors);
+        endpoint.setSubType(subType);
```

Comment:
I think (and you can disagree with that 😄) that the test would be a bit easier to read/maintain if the fields were not declared as `vars` before being set in `endpoint`. Values that are not required could also be omitted. That's only my personal opinion but when I read a test, I like to spend as less time as possible on minor/irrelevant details and focus my brain on the thing that is being tested. This comment also applies to other objects built in this class.
```suggestion
        // Create an endpoint without the type and the properties set.
        final var endpoint = new Endpoint();
        endpoint.setDescription("endpoint-invalid-urls-description");
        endpoint.setName("endpoint-invalid-urls-name");
```

Reply:
Surely it makes it more readable. I think this comes from my habit of writing tests that then check that the values are the same... but this doesn't make sense here. Thanks for the suggestion!